### ISSUE TITLE
valgrind's memcheck tool is not running under Travis

### ISSUE THREAD

**Issue by jodh-intel** [01org/cc-oci-runtime#295]
`.travis.yml` is configured to enable valgrind's `memcheck` tool, but disable `helgrind` and `drd` (see commit 614fb442cba6102f6de7ca44dd02725c7ebca083):

```
$ grep -m 1 valgrind .travis.yml
- ./autogen.sh --enable-cppcheck --enable-valgrind --disable-valgrind-helgrind --disable-valgrind-drd --disable-silent-rules --disable-docker-tests
```

However, looking at the latest Travis log I see:

```
make  -k check-valgrind-memcheck; 
   :
   :
make[1]: Entering directory `/home/travis/build/01org/cc-oci-runtime'
make[1]: *** No rule to make target `check-valgrind-memcheck'.
make[1]: Leaving directory `/home/travis/build/01org/cc-oci-runtime'
make: [check-valgrind] Error 2 (ignored)
```

**Comment by jodh-intel** [comment]
This issue seems to be caused by a bug/limitation in the version of GNU make available in Travis' environment (Ubuntu Trusty/14.04, which provides `make` version 3.81).


The attached makefile is a generalisation of the code generated by `m4/ax_valgrind_check.m4` and demonstrates the problem.

- Running the attached test `Makefile` under make 3.81 results in the following output (incorrect):
```
hello from default
```

- Running the attached test `Makefile` under make 4.1 and 4.2 results in the following output (correct):

```
hello from generated rule for 'memcheck'
hello from generated rule for 'helgrind'
hello from default
```

The simplest workaround is probably to modify `.travis-setup.sh` to build GNU make 4.2 as well.

**Comment by dlespiau** [comment]
:( 

FWIW, I liked the idea of firing the build in a docker image so we don't have to compile those dependencies.

**Comment by jodh-intel** [comment]
Agreed - #305 raised to rework the branch I already have that includes the `Dockerfile`, etc.

**Comment by dcreager** [comment]
I just ran into this same issue on an unrelated project.  Finding this issue helped me stop questioning my sanity!

Returning the favor:  Digging a bit, I found an edit to [ax_valgrind_check.m4](/hst/hst/pull/1/commits/7884aee519e797be79354734a86e636d64f43040) that makes the valgrind checks work on Travis.  Pesky equals signs...

**Comment by jodh-intel** [comment]
Thanks @dcreager!

Re-testing with make 3.81 and a modified version of `test_call_and_foreach.mk.txt` (above) without the equals demonstrates the fix.

[test_call_and_foreach-fixed.mk.txt](https://github.com/01org/cc-oci-runtime/files/561407/test_call_and_foreach-fixed.mk.txt)

**Comment by jodh-intel** [comment]
Fixed by #359.
